### ISSUE TITLE
GUI: HSeparator and VSeparator icons have been swapped

### ISSUE THREAD

**Issue by philippedcote** [godotengine/godot#58469]
### Godot version

3.4.2 Stable

### System information

macOS 12.1 (21C52)

### Issue description

It’s pretty simple. Look at the image to understand. Both icons have been swapped. Just need to be swapped to fix.

<img width="113" alt="image" src="https://user-images.githubusercontent.com/890553/155364191-5655ab32-70e1-4ed6-a766-c4210e452ebe.png">

### Steps to reproduce

1. Insert a VSeparator
2. Insert a HSeparator
3. Notice the icons have been swapped

**Comment by akien-mga** [comment]
I'm pretty sure a PR changed this recently... but I can't find it. Maybe it was another H/V couple.

**Comment by akien-mga** [comment]
So I checked in `master`, and it seems to be that the icons are correct. The problem is likely more the name of the nodes than their icon:
![image](https://user-images.githubusercontent.com/4701338/155372390-f40e2005-c84e-463e-924e-bc928ccb14bd.png)

The description highlights the confusion:

> Vertical version of [Separator]. Even though it looks vertical, it is used to separate objects horizontally.

**Comment by philippedcote** [comment]
No, this is an HSeparator you are seeing. Think of it: This is a separator used in a Horizontal Grid. You put an item next to it to the left and the right, not above and under. ;)

**Comment by akien-mga** [comment]
This was mentioned in https://github.com/godotengine/godot/issues/54161#issuecomment-970364906

>> VSeparator and HSeparator seem to be semantically flipped: I understand that VSeparator stretches visually up and down, but it separates things on the horizontal axis and serves as a child of a horizontal stack. I feel like the visual, in-editor dimensions of the VSeparator are less important for meaningful/consistent naming than the actual purpose it serves---that of being a horizontal separator. Same goes for HSeparator.
>
> Not gonna lie, the current naming makes more sense to me. It's a separator that's vertical. Very self-evident.

With five :+1: on the answer, showing that it's not as obvious as it sounds.

> Alternatively, we can https://github.com/godotengine/godot-proposals/issues/3558 with an enum property that decides the orientation. This approach makes run-time UI layout changes easier too.

That could make sense.

**Comment by akien-mga** [comment]
Checking a couple UI toolkits, they use "vertical divider" the same way we do"VSeparator": the element itself is drawn as a vertical shape, and separates two columns:
https://semantic-ui.com/elements/divider.html
https://getuikit.com/docs/divider

**Comment by Zireael07** [comment]
Maybe mention "vertical divider" in the description then?

**Comment by RefinedSoftwareLLC** [comment]
VListSeparator for inside VBoxContainer,
HListSeparator for inside HBoxContainer?

**Comment by philippedcote** [comment]
> Alternatively, we can [consolidate all those H/V variants into a single node](https://github.com/godotengine/godot-proposals/issues/3558) with an enum property that decides the orientation. This approach makes run-time UI layout changes easier too.

I like this proposal best among all. That would clear up confusion.

In fact, why choose the orientation at all? Can’t the Separator be smart enough to know which orientation it should go depending on context? I would make "Auto" the default behaviour and let H or V be selectable for compatibility purposes.

**Comment by RefinedSoftwareLLC** [comment]
I agree that merging them together with an Auto as default would be best :+1:.

Since that would take coding work, here is a rename idea in the meantime:

RowSeparator for inside VBoxContainer,
ColumnSeparator for inside HBoxContainer?

@pycbouh Could you :+1: this if it does, or :-1: if it doesn't, help clarify between a "vertically drawn line" verses a 'line separating a vertical list'?

**Comment by Calinou** [comment]
> In fact, why choose the orientation at all? Can’t the Separator be smart enough to know which orientation it should go depending on context? I would make "Auto" the default behaviour and let H or V be selectable for compatibility purposes.

The dimension can't *always* be automatically determined, especially in cases where the container's width is equal to its height.

**Comment by philippedcote** [comment]
> The orientation can't _always_ be automatically determined, especially in cases where the container's width is equal to its height.

Indeed. In that case, we should ask the user what to do.

**Comment by pycbouh** [comment]
> @pycbouh Could you 👍 this if it does, or 👎 if it doesn't, help clarify between a "vertically drawn line" verses a 'line separating a vertical list'?

Like I've said on RC, these names could work, but I don't believe there is a problem to begin with. We would be better off updating the documentation as suggested above. There is always going to be somebody who misunderstands something or is confused by something. We should strive to help all, but can only help most, so if the current name work for most, I'd leave it at that.
